src/main/scala/org/apache/spark/sql/TiStrategy.scala

```diff
@@ -74,7 +65,7 @@ class TiStrategy(context: SQLContext) extends Strategy with Logging {
                                    filters: Seq[Expression],
                                    source: TiDBRelation,
                                    selectRequest: TiSelectRequest = new TiSelectRequest): TiSelectRequest = {
-    val tiFilters = filters.map(expr => expr match { case BasicExpression(expr) => expr })
```

Comment:
`map` calls with a single `case` branch is dangerous since it may throw `MatchError` at runtime if this `case` branch doesn't cover all the cases.